### ISSUE TITLE
Returned None Values when computing Gradients

### ISSUE THREAD

**Issue by mhamdan91** [tensorflow/tensorflow#26404]
**Describe the current behavior**
I am using the **Resnet50** code maintained under **_Eager Execution_** as is. When computing gradients 
*grads = tape.gradient(loss, model.variables)
I get NONE values return in grads. I traced down the issue and realized that those none values correspond to the BatchNormalization layers. The layers are set to trainable=True, so I am not quite sure why they do not train or why they return None values. 

******Obviously, model.variables != tape.variables**** 

**Code to reproduce the issue****

_With tf.GradientTape() as tape:
      prediction = model(x, training=True)
      var_list = tape.watched.variables()
loss = compute_loss(prediction, label, operation = 'L2')
grads = tape.gradient(loss, model.variables)
optimizer.apply_gradients(zip(grads, model.variables), global_step = step))_

**Could you please explain why am I getting those None values?**

**Comment by ppwwyyxx** [comment]
You should use `model.trainable_variables` instead of `model.variables`.

**Comment by mhamdan91** [comment]
Thank you @ppwwyyxx , That did get rid of the NONEs. Does this mean BatchNormalization layers hold non-trainable variables?

**Comment by ppwwyyxx** [comment]
> Does this mean BatchNormalization layers hold non-trainable variables?

Yes.
